libs/api/bpf_syscall.cpp

```diff
@@ -262,3 +380,84 @@ bpf(int cmd, union bpf_attr* attr, unsigned int size)
         return -EINVAL;
     }
 }
+
+#define BPF_TO_SYS(field) sys->field = bpf->field
+#define BPF_TO_SYS_STR(field) strncpy_s(sys->field, sizeof(sys->field), bpf->field, _TRUNCATE)
+#define BPF_TO_SYS_MEM(field) memcpy(sys->field, bpf->field, sizeof(sys->field))
+#define SYS_TO_BPF(field) bpf->field = sys->field
+#define SYS_TO_BPF_STR(field) strncpy_s(bpf->field, sys->field, sizeof(bpf->field))
+#define SYS_TO_BPF_MEM(field) memcpy(bpf->field, sys->field, sizeof(bpf->field))
+
+static void
+convert_to_map_info(struct bpf_map_info* bpf, const sys_bpf_map_info_t* sys)
+{
+    SYS_TO_BPF(type);
+    SYS_TO_BPF(id);
+    SYS_TO_BPF(key_size);
+    SYS_TO_BPF(value_size);
+    SYS_TO_BPF(max_entries);
+    SYS_TO_BPF(map_flags);
+    SYS_TO_BPF_STR(name);
+}
+
+static void
+convert_to_sys_map_info(sys_bpf_map_info_t* sys, const struct bpf_map_info* bpf)
+{
+    BPF_TO_SYS(type);
+    BPF_TO_SYS(id);
+    BPF_TO_SYS(key_size);
+    BPF_TO_SYS(value_size);
+    BPF_TO_SYS(max_entries);
+    BPF_TO_SYS(map_flags);
+    BPF_TO_SYS_STR(name);
+}
+
+static void
+convert_to_prog_info(struct bpf_prog_info* bpf, const sys_bpf_prog_info_t* sys)
+{
+    SYS_TO_BPF(type);
+    SYS_TO_BPF(id);
+    // SYS_TO_BPF_MEM(tag);
+    // SYS_TO_BPF(jited_prog_len);
+    // SYS_TO_BPF(xlated_prog_len);
+    // SYS_TO_BPF(jited_prog_insns);
+    // SYS_TO_BPF(xlated_prog_insns);
+    // SYS_TO_BPF(load_time);
+    // SYS_TO_BPF(created_by_uid);
```

Comment:
remove commented out code?

https://github.com/microsoft/ebpf-for-windows/blob/main/docs/DevelopmentGuide.md#coding-conventions says:
> **DON'T** use commented-out code, or code in an #if 0 or equivalent. Make sure all code is actually built.